frontend/src/app/projects/models/model-source/t4c/manage-t4c-model/manage-t4c-model.component.ts

```diff
@@ -0,0 +1,200 @@
+      (instance: T4CInstance) => instance.id == this.form.value.t4cRepositoryId
+    );
+  }
+
+  t4cRepositories: T4CRepository[] | undefined = undefined;
+  get selectedRepository(): T4CRepository | null | undefined {
+    return this.t4cRepositories?.find(
+      (repository: T4CRepository) =>
+        repository.id == this.form.value.t4cRepositoryId
+    );
+  }
+
+  t4cModel?: T4CModel;
+
+  public form = new FormGroup({
+    t4cInstanceId: new FormControl<null | number>(null, [
+      Validators.required,
+      notNullValidator(),
+    ]),
+    t4cRepositoryId: new FormControl<null | number>(null, [
+      Validators.required,
+      notNullValidator(),
+    ]),
+    name: new FormControl('', [Validators.required]),
+  });
+
+  constructor(
+    public projectService: ProjectService,
+    public modelService: ModelService,
+    public t4cInstanceService: T4CInstanceService,
+    public t4cRepositoryService: T4CRepoService,
+    public t4cModelService: T4CModelService,
+    private router: Router,
+    private route: ActivatedRoute,
+    private toastService: ToastService
+  ) {}
+
+  ngOnInit(): void {
+    this.form.disable({ emitEvent: false });
+    this.loading = true;
+    this.updateRepositoryListOnInstanceChange();
+    this.updateProjectListOnRepositoryChange();
+
+    this.fetchT4CInstances()
+      .pipe(switchMap(() => this.patchFormWithExistingT4CModel()))
+      .subscribe(() => {
+        this.loading = false;
+      });
+  }
+
+  fetchT4CInstances(): Observable<T4CInstance[]> {
+    return this.t4cInstanceService.listInstances().pipe(
+      tap((instances) => {
+        this.t4cInstances = instances;
+        this.form.controls.t4cInstanceId.enable({ emitEvent: false });
+        this.form.controls.t4cInstanceId.patchValue(
+          this.form.controls.t4cInstanceId.value
+        );
+      })
+    );
+  }
+
+  updateRepositoryListOnInstanceChange(): void {
+    this.form.controls.t4cInstanceId.valueChanges.subscribe(
+      (t4cInstanceId: number | null) => {
+        if (t4cInstanceId) {
+          this.t4cRepositoryService
+            .getT4CRepositories(t4cInstanceId)
+            .subscribe((repositories) => {
+              this.t4cRepositories = repositories;
+              this.form.controls.t4cRepositoryId.enable({
+                emitEvent: false,
+              });
+            });
+        }
+      }
+    );
+  }
+
+  updateProjectListOnRepositoryChange(): void {
+    this.form.controls.t4cRepositoryId.valueChanges.subscribe(() => {
+      this.form.controls.name.enable({ emitEvent: false });
+    });
+  }
+
+  patchFormWithExistingT4CModel(): Observable<T4CModel | undefined> {
+    return this.t4cModelService._t4cModel.pipe(
+      tap((model: T4CModel | undefined) => {
+        if (model) {
+          this.editing = true;
+          this.t4cModel = model;
+
+          this.form.controls.t4cInstanceId.setValue(
+            model.repository.instance.id,
+            { emitEvent: true }
+          );
```

Comment:
```suggestion
          this.form.controls.t4cInstanceId.setValue(
            model.repository.instance.id
          );
```
True is the default case here.